File: ride-management-service/src/models/Ride.js
const mongoose = require("mongoose");

const rideSchema = new mongoose.Schema(
{
    passengerId: {
        type: String,
        required: true
    },

    driverId: {
        type: String
    },

    pickupLocation: {
        type: String,
        required: true
    },

    destination: {
        type: String,
        required: true
    },

    status: {
        type: String,
        enum: [
            "REQUESTED",
            "ASSIGNED",
            "ACCEPTED",
            "IN_PROGRESS",
            "COMPLETED",
            "CANCELLED"
        ],
        default: "REQUESTED"
    },

    estimatedFare: {
        type: Number,
        default: 0
    },

    finalFare: {
        type: Number,
        default: 0
    }
},
{
    timestamps: true
}
);

module.exports = mongoose.model("Ride", rideSchema);

File: ride-management-service/src/main/java/com/ridelink/ride_management_service/model/Ride.java
package com.ridelink.ride_management_service.model;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "rides")
@Data
public class Ride {

    @Id
    private String id;

    private String passengerId;

    private String driverId;

    private String pickupLocation;

    private String destination;

    private String status = "REQUESTED";

    private Double estimatedFare = 0.0;

    private Double finalFare = 0.0;
}